# Practical: Bioinformatic follow-up of GWAS hits
# Objectives
Do some bioinformatic follow-up investigation of GWAS hits using a range of online tools

1. Generate a locuszoom plot to visualise the association signal
2. See if the association is known or novel 
3. Identify interesting annotations using Haploreg
4. Use the ensembl genome browser to explore the region and visualise annotations

#Data
This practical uses results generated in the GWAS session - see data saved in exercise 3

Regional association results files have already been generated and saved here: `GenEpi_shortcourse_2016\Day2_Section6_GWAS`

We will access external resources through websites


#Exercise 1 - Generate locuszoom plots
Choose a signal to investigate: rs571312; rs630372; rs8050136; rs13130484; rs2867125; rs12748679

Go to [locuszoom website](http://locuszoom.sph.umich.edu/locuszoom/genform.php?type=yourdata)

Upload the regional association results (for the relevant chromosome) already prepared for you, e.g. `chr18_assoc.txt`

`The p-value column is 'P'`

`The marker column is 'SNP'`

`The delimiter is 'whitespace'`

`Enter your SNP name into the box.`


Now, create your plot!

**_Questions:_**
> (1) Is there much resolution for the signal (many or few SNPs, small or large region)?

> (2) Are all associated SNPs in high LD?

> (3) What genes are nearby?


#Exercise 2 - GWAS catalog
See if the association with your SNP (gene or region) is novel or known

Go to the [EBI GWAS catalog website](https://www.ebi.ac.uk/gwas/)

**_Question:_**
> (4) Is this association with BMI known or novel?


#Exercise 2 - Investigating your signal in Haploreg
We will now test if our signal overlaps with any interesting annotations:

- coding variant
- eQTL results
- functional annotations in ENCODE/RoadMap Epigenomics

Open the [Haploreg website](http://www.broadinstitute.org/mammals/haploreg/haploreg.php)

There are different ways to select your 'credible SNP set'. Here we will just look up everything with r<sup>2</sup>>0.8 with top SNP.

`Enter your SNP name into the box and click submit`

**_Questions:_**
> (5) Are any SNPs coding SNPs? (see last column)

> (6) Are any SNPs known eQTLs? If so for what gene? In what tissue?

> (7) Could you say with any confidence which is the causal variant?

> (8) Could you say with any confidence which is the causal gene?

> (9) Could you say with any confidence which is the tissue of interest?


**_Extra Questions:_**
> (10) How many have histone marks indicative of a promoter? an enhancer? In a region of open chromatin? Have proteins bound?

> (11) Which SNP alters the most motifs? Which motif is most affected?



#Exercise 3 - Exploring the region in ensembl
Now we will explore the same region in ensembl & see visualisation of the same regulatory data.

Go to the [Ensembl website](http://grch37.ensembl.org/Homo_sapiens/Info/Index)

`Enter your SNP name`

`Click on some of the 'Explore this variant' boxes to see what sort of information you can get`

`Click on the Location tab to visualise the region`

`Zoom to view around 50kb`

`Note the 1KG - All SNPs/INDELs track - use the key to work out the variant types`

`Note the Multi Cell Regulatory Features column - click on some marks`

`Click on 'Configure this page' & Enable all regulatory features`


#Summary
For your region:

> Is the association novel?

> What are the nearby genes?

> Is there any evidence for a particular .....

> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; causal variant?

>  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; causal gene?

> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; affected tissue?

> What additional investigations/validation might you want to do?


