# Practical: Bioinformatic follow-up of GWAS hits
# Objectives
Do some bioinformatic follow-up investigation of GWAS hits using a range of online tools

1. Generate a locuszoom plot to visualise the association signal
3. Identify interesting annotations using Haploreg
4. 2. Use the ensembl genome browser to explore the region and visualise annotations

#Data
This practical uses results generated in the GWAS session - see data saved in exercise 3

Regional association results files have already been generated and saved here: `GenEpi_shortcourse_2016\Day2_Section6_GWAS`

We will access external resources through websites


#Exercise 1 - Generate locuszoom plots
Choose a signal to investigate: rs571312; rs630372; rs8050136; rs13130484; rs2867125

Go to [locuszoom website](http://locuszoom.sph.umich.edu/locuszoom/genform.php?type=yourdata)

Upload the regional association results (for the relevant chromosome) already prepared for you, e.g. `chr18_assoc.txt`

`The p-value column is 'P'`

`The marker column is 'SNP'`

`The delimiter is 'whitespace'`

`Enter your SNP name into the box.`


Now, create your plot!

**_Questions:_**
> (1) How many SNPs are significant in this region?

> (2) Are they all in high LD?

> (3) Over what region does the association span? (in bp)

> (4) What genes are nearby?


#Exercise 2 - Investigating your signal in Haploreg
We will now test if our signal overlaps with any interesting annotations:
- Previous associations
- eQTL results
- functional annotations in ENCODE/RoadMap Epigenomics

Open the [Haploreg website](http://www.broadinstitute.org/mammals/haploreg/haploreg.php)

There are different ways to select your 'credible SNP set'. Here we will just look up everything with r<sup>2</sup>>0.8 with top SNP.

`Enter your SNP name into the box and click submit`

**_Question:_**
> (5) What other traits have been associated in this region? (look at NHGRI/EBI GWAS hits column + click on SNPs to see trait)

> (6) Are any SNPs known eQTLs? For what gene? In what tissue?

> (7) How many have histone marks indicative of a promoter? an enhancer? In a region of open chromatin? Have proteins bound?

> (8) Which SNP alters the most motifs? Which motif is most affected?


#Exercise 3 - Exploring the region in ensembl
Now we will explore the same region in ensembl
Go to the [Ensembl website](http://grch37.ensembl.org/Homo_sapiens/Info/Index)

`Enter your SNP name`

`Click on some of the 'Explore this variant' boxes to see what sort of information you can get`

`Click on the Location tab to visualise the region`

`Zoom to view around 50kb`

`Note the 1KG - All SNPs/INDELs track - use the key to work out the variant types`

`Note the Multi Cell Regulatory Features column - click on some marks`

`Click on 'Configure this page' & Enable all regulatory features`

**_Questions:_**

Pick either the top SNP or one that has stood out for you
> (9) According to ensembl what type of variant is this?

Find the nearest regulatory region

> (10) What is the regulatory annotation? 

> (11) What cell types is this annotation in?


